public/apps/configuration/panels/audit-logging/audit-logging.tsx

```diff
@@ -16,50 +16,36 @@
 import React, { useEffect, useState } from 'react';
 
 import {
-  EuiDescribedFormGroup,
-  EuiForm,
-  EuiFormRow,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
   EuiHorizontalRule,
   EuiPanel,
-  EuiSwitch,
+  EuiSpacer,
   EuiTitle,
 } from '@elastic/eui';
 import { AppDependencies } from '../../../types';
 import { API_ENDPOINT_AUDITLOGGING } from '../../constants';
-
-function renderStatusPanel(onSwitchChange: any, auditLoggingEnabled: boolean) {
-  return (
-    <EuiForm>
-      <EuiDescribedFormGroup
-        title={<h3>Enable audit logging</h3>}
-        description={<>Enable or disable audit logging</>}
-      >
-        <EuiFormRow>
-          <EuiSwitch
-            name="auditLoggingEnabledSwitch"
-            label={auditLoggingEnabled ? 'Enabled' : 'Disabled'}
-            checked={auditLoggingEnabled}
-            onChange={onSwitchChange}
-          />
-        </EuiFormRow>
-      </EuiDescribedFormGroup>
-    </EuiForm>
-  );
-}
+import {
+  renderComplianceSettings,
+  renderGeneralSettings,
+  renderStatusPanel,
+} from '../../utils/audit-logging-view-utils';
 
 export function AuditLogging(props: AppDependencies) {
   const [auditLoggingEnabled, setAuditLoggingEnabled] = useState<boolean>(false);
-
+  const [configuration, setConfiguration] = useState({});
   const onSwitchChange = () => {
     setAuditLoggingEnabled((prevAuditLoggingEnabled) => !prevAuditLoggingEnabled);
   };
 
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const configuration = await props.coreStart.http.get(API_ENDPOINT_AUDITLOGGING);
+        const rawConfiguration = await props.coreStart.http.get(API_ENDPOINT_AUDITLOGGING);
 
-        setAuditLoggingEnabled(configuration.data.config.enabled);
+        setAuditLoggingEnabled(rawConfiguration.data.config.enabled);
+        setConfiguration(rawConfiguration.data.config);
```

Comment:
Why we need the states for the whole thing `config` and also part of if `config.enabled`?

Reply:
Currently enable/disable hasn't called the post request yet. I set it for viewing the switch change. Once the post call is made, I can just make them use config.enabled.